project-overview.md

```diff
@@ -0,0 +1,146 @@
+
+When a node commits to a storage contract and a user uploads a file or other arbitrary data, the network will proactively verify that the storing node is online and the data is retrievable. Storage nodes broadcast proofs of data possession over random intervals. If the storage node sent invalid proofs or failed to provide them in time, the network will re-post the contract for any other storage node to pick up. When the contract is re-posted, an amount from the faulty node's stake is used for the new storing node bandwidth fees. It is expected that data is stored on at least several nodes to prevent data loss in the case of a catastrophic network failure. Erasure coding complements active verification by allowing to reconstruct the full set from a subset of the data.
+
+### Proofs of data possession and retrievability
+
+We use proofs of data possession and retrievability to ensure storage nodes committed to a contract remain online and available. The storage and retrievability proofs are formally described in this [document](https://hackmd.io/2uRBltuIT7yX0CyczJevYg?view).
+
+The main objective of the proofs are:
+
+- Ensure nodes are online and maintaining the entirety of the dataset from the storage contract
+- Ensure that data is readily retrievable to prevent blackmailing and withholding attacks
+
+
+## Interacting with the Dagger Network
+
+Any regular node that participates in the network needs to have an operational amount set aside in order to cover for bandwidth fees. This creates a barrier to entry however, we think it's a worthy tradeoff in order to maintain the security and health of the network. It's worth noting that any decentralized platform will have similar requirements and limitations. Bellow, we'll list some potential ways to workaround this in Dagger.
+
+### Subsidies or airdrops
+
+Any application migrating or being built for a deserialized platform requires some operational capital to participate. Many projects workaround this by initially subsidizing potential users with small portions of their token. This are usually known as airdrops. Dagger can use a similar technique to allow first adopters to begin participating in the network.
+
+### Tit-for-tat settlements
+
+Many interactions in the network will be long lived, this allows two nodes to exchange in a tit-for-tat manner. It works like this, a long lived payment channel is opened and nodes freely exchange chunks regardless of which way the balance tilts. The channel can only be closed when both parties agree (this is always true, regardless of how long the channel has been open). The node that is currently in debt will need to add funds to the channel or keep providing the other peer with chunks until the debt is repaid. The channel is closed only when the debt is completely settled or the counter party forgoes it.  We expect that nodes resort to tit-for-tat often, thus alleviating the need to constantly "top up" the node's balance.
```

Comment:
That's probably reasonable, but it doesn't change the mechanics described here much, or does it?

Reply:
Yes, the tit-for-tat scheme still works, only the statement that channel can only be closed when both parties agree is not entirely correct.

Reply:
I think of this more as a fallback mechanism rather than a fundamental property tho?

Reply:
Maybe worth clarifying, but in this context I'm using payment channels as a placeholder to describe micropayments, rather than any specific implementation.